src/v/kafka/server/group.cc

```diff
@@ -1244,6 +1276,28 @@ group::store_offsets(offset_commit_request&& r) {
       });
 }
 
+ss::future<txn_offset_commit_response>
+group::handle_txn_offset_commit(txn_offset_commit_request&& r) {
+    if (in_state(group_state::dead)) {
+        co_return txn_offset_commit_response(
+          r, error_code::coordinator_not_available);
+    } else if (in_state(group_state::empty)) {
+        // <kafka>The group is only using Kafka to store offsets.</kafka>
+        co_return co_await store_txn_offsets(std::move(r));
+    } else if (
+      in_state(group_state::stable)
+      || in_state(group_state::preparing_rebalance)) {
+        co_return co_await store_txn_offsets(std::move(r));
+    } else if (in_state(group_state::completing_rebalance)) {
+        co_return txn_offset_commit_response(
+          r, error_code::rebalance_in_progress);
+    } else {
+        vlog(klog.error, "Unexpected group state {} for {}", _state, *this);
+        co_return txn_offset_commit_response(
+          r, error_code::unknown_server_error);
```

Comment:
i wonder if we can be more specific in this error. we know a few things already. 

1. we know that it's not `stable, rebalance, preparing, empty or dead` 
2. we also know this happened inside the group::handle_txn... which presumably passed some checks already

not sure what enum make sense, but my hunch is that something more specific might help debugging.

Reply:
it follows the pattern set by handle_leave_group, and handle_offset_commit so I don't really want to do something different or change the scope of the PR by refactoring existing code. As I understand this else branch is precautions against adding new enum values without updating all the switch branches.